### ISSUE TITLE
Ansible-lint should catch invalid attributes

### ISSUE THREAD

**Issue by MaciejKucia** [ansible-community/ansible-lint#539]
# Issue Type
- Feature request

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.5.15
ansible-lint --version
ansible-lint 4.0.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

During Ansible play linted playbooks and roles should not report following warning:
```
[WARNING]: Ignoring invalid attribute: backup
```

Sample invalid part of role:
```
- name: Configure nfs
  backup: yes
  template:
    src: nfs.j2
    dest: /etc/nfs
```

**Comment by evitalis** [comment]
That should be an error as backup needs to be under template anyways for it to be valid. Is your request for an error instead of warning?

**Comment by MaciejKucia** [comment]
```
$ pip freeze
ansible==2.8.3
ansible-lint==4.1.0
asn1crypto==0.24.0
cffi==1.12.3
cryptography==2.7
enum34==1.1.6
ipaddress==1.0.22
Jinja2==2.10.1
MarkupSafe==1.1.1
pycparser==2.19
PyYAML==5.1.2
ruamel.ordereddict==0.4.14
ruamel.yaml==0.16.1
ruamel.yaml.clib==0.1.2
six==1.12.0

$ ansible-lint playbook.yml; echo $?
0

$ cat playbook.yml 
- hosts: all
  tasks:
    - name: Configure nfs
      backup: yes
      template:
        src: nfs.j2
        dest: /etc/nfs
```

My request is for any feedback (warning or error) from linter :wink:

**Comment by evitalis** [comment]
I think what you want is `ansible-playbook --syntax-check' not from the linter in this case. My understanding is a longer is for formatting and best practices not nessesarily error catching.

**Comment by MaciejKucia** [comment]
Maybe ansible-lint should call `ansible-playbook --syntax-check` underneath? 
What is the point of running linter if syntax errors are not detected?

**Comment by ssbarnea** [comment]
v5.0.0 does exactly this.
